src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -40,21 +47,33 @@ public SnapshotBuilderGetConceptsResponse getConceptChildren(UUID datasetId, Int
                     .id(conceptId + 1)));
   }
 
-  private int getRollupCount(UUID datasetId, List<SnapshotBuilderCohort> cohorts) {
-    return 100;
-  }
-
   public SnapshotBuilderCountResponse getCountResponse(
       UUID id, List<SnapshotBuilderCohort> cohorts) {
     return new SnapshotBuilderCountResponse()
         .sql("")
-        .result(new SnapshotBuilderCountResponseResult().total(getRollupCount(id, cohorts)));
+        .result(
+            new SnapshotBuilderCountResponseResult()
+                .total(
+                    getRollupCountForCriteriaGroups(
+                        id,
+                        cohorts.stream()
+                            .map(SnapshotBuilderCohort::getCriteriaGroups)
+                            .collect(Collectors.toList()))));
   }
 
   public EnumerateSnapshotAccessRequest enumerateByDatasetId(UUID id) {
     return convertToEnumerateModel(snapshotRequestDao.enumerateByDatasetId(id));
   }
 
+  public int getRollupCountForCriteriaGroups(
+      UUID datasetId, List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList) {
+    Query query =
+        new CriteriaQueryBuilder()
+            .generateRollupCountsQueryForCriteriaGroupsList(criteriaGroupsList);
+    query.renderSQL();
+    return 5;
```

Comment:
when do we plan to remove the stubbed response?

Reply:
I need to rebase to move Shelby's code in, and then I should be able to remove it now.